perf_test/sparse/KokkosSparse_block_pcg.cpp

```diff
@@ -0,0 +1,743 @@
+      KokkosKernels::Impl::copy_vector<INDEX_TYPE * , row_map_view_t, myExecSpace>(nv+1, xadj, rowmap_view);
+
+      graph_t static_graph (columns_view, rowmap_view);
+      crsMat_t crsmat("CrsMatrix", nv, values_view, static_graph);
+      delete [] xadj;
+      delete [] adj;
+      delete [] ew;
+
+      values_view_t kok_x_original = create_x_vector<values_view_t>(((nv /block_size) + 1) * block_size, MAXVAL);
+      for (INDEX_TYPE i = nv; i < ((nv /block_size) + 1) * block_size; ++i){
+    	  kok_x_original(i) = 0;
+      }
+      run_experiment<myExecSpace, crsMat_t>(crsmat, kok_x_original, block_size);
+
+      myExecSpace::finalize();
+    }
+
+#endif
+
+#if defined( KOKKOS_HAVE_OPENMP )
+
+    if ( cmdline[ CMD_USE_OPENMP ] ) {
+      INDEX_TYPE nv = 0, ne = 0;
+      INDEX_TYPE *xadj, *adj;
+      SCALAR_TYPE *ew;
+
+
+      if ( cmdline[ CMD_USE_NUMA ] && cmdline[ CMD_USE_CORE_PER_NUMA ] ) {
+        Kokkos::OpenMP::initialize( cmdline[ CMD_USE_OPENMP ] ,
+                                     cmdline[ CMD_USE_NUMA ] ,
+                                     cmdline[ CMD_USE_CORE_PER_NUMA ] );
+      }
+      else {
+        Kokkos::OpenMP::initialize( cmdline[ CMD_USE_OPENMP ] );
+      }
+      Kokkos::OpenMP::print_configuration(std::cout);
+
+      KokkosKernels::Impl::read_matrix<INDEX_TYPE,INDEX_TYPE, SCALAR_TYPE> (&nv, &ne, &xadj, &adj, &ew, mtx_bin_file);
+
+
+      typedef Kokkos::OpenMP myExecSpace;
+      typedef typename KokkosSparse::CrsMatrix<SCALAR_TYPE, INDEX_TYPE, myExecSpace, void, SIZE_TYPE > crsMat_t;
+
+      typedef typename crsMat_t::StaticCrsGraphType graph_t;
+      typedef typename crsMat_t::row_map_type::non_const_type row_map_view_t;
+      typedef typename crsMat_t::index_type::non_const_type   cols_view_t;
+      typedef typename crsMat_t::values_type::non_const_type values_view_t;
+
+      row_map_view_t rowmap_view("rowmap_view", nv+1);
+      cols_view_t columns_view("colsmap_view", ne);
+      values_view_t values_view("values_view", ne);
+
+      KokkosKernels::Impl::copy_vector<SCALAR_TYPE * , values_view_t, myExecSpace>(ne, ew, values_view);
+      KokkosKernels::Impl::copy_vector<INDEX_TYPE * , cols_view_t, myExecSpace>(ne, adj, columns_view);
+      KokkosKernels::Impl::copy_vector<INDEX_TYPE * , row_map_view_t, myExecSpace>(nv+1, xadj, rowmap_view);
+
+      graph_t static_graph (columns_view, rowmap_view);
+      crsMat_t crsmat("CrsMatrix", nv, values_view, static_graph);
+
+      //crsMat_t crsmat("CrsMatrix", nv, nv, ne, ew, xadj, adj);
+      delete [] xadj;
+      delete [] adj;
+      delete [] ew;
+
+
+      values_view_t kok_x_original = create_x_vector<values_view_t>(((nv /block_size) + 1) * block_size, MAXVAL);
+      for (INDEX_TYPE i = nv; i < ((nv /block_size) + 1) * block_size; ++i){
+    	  kok_x_original(i) = 0;
+      }
+      run_experiment<myExecSpace, crsMat_t>(crsmat, kok_x_original, block_size);
+
+      myExecSpace::finalize();
```

Comment:
I think calling finalize() for a specific execution space is to be deprecated in Kokkos, @crtrott.